04.ls/ls.rb

```diff
@@ -33,4 +59,93 @@ def split_files(files)
   divided_files
 end
 
+def show_long_format(path, file_names)
+  stats_by_path = file_names.to_h do |file_name|
+    absolute_path = File.join(path, file_name)
+    [absolute_path, File.lstat(absolute_path)]
+  end
+  max_lengths = result_max_lengths(stats_by_path.values)
+
+  puts "total #{stats_by_path.values.sum(&:blocks)}"
+  generate_file_data_list(stats_by_path).each do |file_data|
+    puts format_file(file_data, max_lengths)
+  end
+end
+
+def generate_file_data_list(stats_by_path)
+  stats_by_path.map do |absolute_path, stat|
+    {
+      permission: to_permission(stat),
+      link_count: stat.nlink.to_s,
+      owner: Etc.getpwuid(stat.uid).name,
+      group: Etc.getgrgid(stat.gid).name,
+      byte_size: stat.size.to_s,
+      time_stamp: stat.mtime.strftime(to_timestamp(stat.mtime)),
```

Comment:
```suggestion
      time_stamp: to_timestamp(stat.mtime),
```

としてメソッド側で出力内容をフォーマットしてしまうといいのでは

Reply:
確かに`stat.mtime`が重複していて見栄えも悪いですね。修正しました。